aiojobs/_job.py

```diff
@@ -14,9 +14,15 @@
 
 
 class Job(Generic[_T]):
-    def __init__(self, coro: Coroutine[object, object, _T], scheduler: Scheduler):
+    def __init__(
+        self,
+        coro: Coroutine[object, object, _T],
+        scheduler: Scheduler,
+        name: Optional[str] = None,
+    ):
         self._coro = coro
         self._scheduler: Optional[Scheduler] = scheduler
+        self._name: Optional[str] = name
```

Comment:
asyncio.Task has this:
```
        if name is None:
            self._name = f'Task-{_task_name_counter()}'
```

So, maybe we should do something similar:
```suggestion
        if name is None:
            self._name = "Job({})".format(coro)
        else:
            self._name = name
```
Or maybe you have a better idea?

Reply:
My thought behind leaving it "None if not set" was that it would have the lowest potential of breaking anyone who may already be relying on the default Task names for something. 

By passing None down into the create_task function this new Job Name functionality becomes 100% opt in. Calling job.name (or Job.get_name()) on a Job whose name was not explicitly set will return back the Task-<task counter> name.

Reply:
Ah, yes, we don't want to set it here, you're right.

But, we could use it as a default in `get_name()` to ensure it always returns a `str`, same as `Task.get_name()`.